File: javalint-ec4j-linter/src/main/java/io/github/kamilperczynski/javalint/ec4j/linters/NoopFormattingEvents.java
package io.github.kamilperczynski.javalint.ec4j.linters;

import io.github.kamilperczynski.javalint.formatter.FormatterEvents;
import org.jetbrains.annotations.NotNull;

import java.nio.file.Path;

public enum NoopFormattingEvents implements FormatterEvents {

  INSTANCE;

  @Override
  public void formattingStarted() {
  }

  @Override
  public void fileFormattingStarted(@NotNull Path path) {
  }

  @Override
  public void fileIgnored(@NotNull Path path) {
  }

  @Override
  public void fileFormattingEnd(@NotNull Path path, boolean isModified) {
  }

  @Override
  public void formattingEnd() {
  }

}

File: javalint-formatter/src/main/java/io/github/kamilperczynski/javalint/formatter/NoopFormattingEvents.kt
package io.github.kamilperczynski.javalint.formatter

import java.nio.file.Path

enum class NoopFormattingEvents : FormatterEvents {

  INSTANCE;

  override val reformattedFilesCount: Int get() = 0

  override fun formattingStarted() {
  }

  override fun fileFormattingStarted(path: Path) {
  }

  override fun fileIgnored(path: Path) {
  }

  override fun fileFormattingEnd(path: Path, isModified: Boolean) {
  }

  override fun formattingEnd() {
  }

}

